### ISSUE TITLE
Multiple OLED devices with ESP32

### ISSUE THREAD

**Issue by thijstriemstra** [WifWaf/TCA9548A#4]
I'm trying to use multiple 0.91 inch OLED displays (128x32px) with an ESP32 board and a TCA9548A. I found this: https://gist.github.com/johnaboxall/653cc12aa66571b6490610c20ac3b7c8 and it seems like a good start but I'd rather use a library like this. Can you show how to port that example using this library?

**Comment by WifWaf** [comment]
[Something like this](https://pastebin.com/8tnW7jEj)?

I can't test it, but it should give you an idea.

**Comment by thijstriemstra** [comment]
thanks I'll give it a try when all devices arrived and report back.

Pasting content here in case that page gets deleted:

```cpp
#include <Arduino.h>
#include <U8g2lib.h>
#include <Wire.h>
#include "TCA9548A.h"
 
TCA9548A I2CMux(0x70); 
U8G2_SSD1306_64X32_1F_F_HW_I2C u8g2(U8G2_R0, /* reset=*/ U8X8_PIN_NONE);
 
void setup() 
{
  I2CMux.begin(Wire);             // TwoWire isntance can be passed here as 3rd argument
  I2CMux.closeAll();              // Set a base state which we know (also the default state on power on)  
  I2CMux.openChannel(0);          // tell the TCA9548A we want to use I2C bus number zero (to talk to the OLED)
 
  u8g2.begin();
}
 
void loop()
{
   // use the OLED as normal
    u8g2.clearBuffer();          		// clear the internal memory
    u8g2.setFont(u8g2_font_inb24_mr );  // choose a suitable font
    u8g2.setCursor(0, 24);
    u8g2.print(a);
    a = a - 47;
    u8g2.sendBuffer();          		// transfer internal memory to the display
    delay(100);
}
```

**Comment by thijstriemstra** [comment]
I tested with 2 1.3" SH1106 128x64 OLEDs and this library: https://github.com/ThingPulse/esp8266-oled-ssd1306

```cpp
#include <Arduino.h>
#include <Wire.h>
#include <TCA9548A.h>
#include <SH1106Wire.h>

int displayWidth = 128;
int displayHeight = 64;

TCA9548A I2CMux(0x70); 
TwoWire twire = TwoWire(1);
SH1106Wire display(OLEDAddress, I2CMuxSDAPin, I2CMuxSCLPin, GEOMETRY_128_64, I2C_TWO);

void setup() {
  Serial.begin(115200);

  I2CMux.begin(twire, I2CMuxSDAPin, I2CMuxSCLPin);
  I2CMux.closeAll();

  // draw on screen 1
  I2CMux.openChannel(0);
  display.init();
  display.flipScreenVertically();
  display.clear();
  display.setFont(ArialMT_Plain_24);
  display.setTextAlignment(TEXT_ALIGN_CENTER);
  display.drawRect(0, 0, displayWidth, displayHeight);
  display.drawString(64, 20, "0");
  display.display();
  I2CMux.closeChannel(0);

  // draw on screen 2
  I2CMux.openChannel(1);
  display.init();
  display.flipScreenVertically();
  display.clear();
  display.setFont(ArialMT_Plain_24);
  display.setTextAlignment(TEXT_ALIGN_CENTER);
  display.drawRect(0, 0, displayWidth, displayHeight);
  display.drawString(64, 20, "1");
  display.display();
  I2CMux.closeAll();

  Serial.println("Started.");
}

void loop() {
}
```

It needs the fix for #5 though. Seems to work!
